cfme/configure/configuration/region_settings.py

```diff
@@ -428,3 +431,344 @@ class MapTagsEdit(CFMENavigateStep):
+            'url': self.url,
+            'username': self.username,
+            'password': self.password,
+            'password_verify': password_verify,
+            'repo_name': self.repo_name,
+            'use_proxy': self.use_proxy,
+            'proxy_url': self.proxy_url,
+            'proxy_username': self.proxy_username,
+            'proxy_password': self.proxy_password,
+            'proxy_password_verify': proxy_password_verify
+        }
+
+        view.fill(details)
+
+        if self.set_default_rhsm_address:
+            view.rhn_default_url.click()
+
+        if self.set_default_repository:
+            view.repo_default_name.click()
+
+        if validate:
+            view.validate_button.click()
+
+        if cancel:
+            view.cancel_button.click()
+            flash_message = 'Edit of Customer Information was cancelled'
+        else:
+            view.save_button.click()
+            flash_message = 'Customer Information successfully saved'
+
+        view = self.create_view(RedHatUpdatesView)
+        view.flash.assert_message(flash_message)
+
+    def refresh(self):
+        """ Click refresh button to update statuses of appliances """
+        view = navigate_to(self, 'Details')
+        view.refresh.click()
+
+    def register_appliances(self, *appliance_names):
+        """ Register appliances by names
+
+        Args:
+            appliance_names: Names of appliances to register; will register all if empty
+        """
+        view = navigate_to(self, 'Details')
+        self.select_appliances(*appliance_names)
+        view.register.click()
+        view.flash.assert_message("Registration has been initiated for the selected Servers")
+
+    def update_appliances(self, *appliance_names):
+        """ Update appliances by names
+
+        Args:
+            appliance_names: Names of appliances to update; will update all if empty
+        """
+        view = navigate_to(self, 'Details')
+        self.select_appliances(*appliance_names)
+        view.apply_updates.click()
+        view.flash.assert_message("Update has been initiated for the selected Servers")
+
+    def check_updates(self, *appliance_names):
+        """ Run update check on appliances by names
+
+        Args:
+            appliance_names: Names of appliances to check; will check all if empty
+        """
+        view = navigate_to(self, 'Details')
+        self.select_appliances(*appliance_names)
+        view.check_updates.click()
+        view.flash.assert_message("Check for updates has been initiated for the selected Servers")
+
+    def is_registering(self, *appliance_names):
+        """ Check if at least one appliance is registering """
+        view = navigate_to(self, 'Details')
+        for appliance_name in appliance_names:
+            row = view.updates_table.row(appliance=appliance_name)
+            if row.last_message.text.lower() == 'registering':
+                return True
+            else:
+                return False
```

Comment:
**REQUIRED**
A logic mistake is here. If the first appliance is not registering other ones won't be checked.
```python
for appliance_name in appliance_names:
    row = view.updates_table.row(appliance=appliance_name)
    if row.last_message.text.lower() == 'registering':
        return True
else:
    return False
```